UefiTestingPkg/AuditTests/PagingAudit/Windows/MemoryRangeObjects.py

```diff
@@ -135,10 +147,52 @@ def GetSystemMemoryType(self):
         if self.SystemMemoryType is None:
             return "None"
         try:
+            # Indices 5, 6, and 7 corrolate with a formatted string in the SystemMemoryTypes list.
+            # The formatted string includes a place for the CPU number of the memory range object, so
+            # if it is one of indices 5, 6, or 7 then return it as a formatted string with the CPU number.
+            if (self.SystemMemoryType == 5 or self.SystemMemoryType == 6 or self.SystemMemoryType == 7):
+                return MemoryRange.SystemMemoryTypes[self.SystemMemoryType].format(self.CpuNumber)
+
             return MemoryRange.SystemMemoryTypes[self.SystemMemoryType]
         except:
             raise Exception("System Memory Type is invalid %d" % self.SystemMemoryType)
 
+    def StackInit (self, SystemMemoryType, PhysicalStart, Length):
+        self.PhysicalStart = PhysicalStart
+        self.PhysicalSize = Length
+
+        # Convert the system type to an index in the object list of memory types
+        if SystemMemoryType == "StackGuard":
+            self.SystemMemoryType = 3

```

Comment:
nit: Would having SystemMemoryType as an enum be better for readability? I.e. SystemMemoryType.StackGuard, etc.

Reply:
Please take another look and tell me if the update is what you had a mind

Reply:
Looks better to me

Reply:
LGTM